tests/test_optimizer/test_dist_came.py

```diff
@@ -1,308 +1,573 @@
-    tp_group = proc_mesh.get_group_along_axis(0)
-    dp_group = proc_mesh.get_group_along_axis(1)
-    tp_rank = dist.get_rank(tp_group)
+    tp_group, dp_group = proc_mesh.get_group_along_axis(0), proc_mesh.get_group_along_axis(1)
 
-    seed_all(_SEED)
-    clear_layout_converter()  # Ensure correct sharding
-    torch_model = Net(_IN_DIM, _HID_DIM).to(rank)
-    tp_model = TPNet(torch_model.fc0, torch_model.fc1, torch_model.fc2, tp_group).to(rank)
-
-    assert_close(
-        torch_model.fc1.weight[tp_rank * _HID_DIM // tp_size : (tp_rank + 1) * _HID_DIM // tp_size],
-        tp_model.fc1.weight,
-    )
-    assert_close(
-        torch_model.fc2.weight[:, tp_rank * _HID_DIM // tp_size : (tp_rank + 1) * _HID_DIM // tp_size],
-        tp_model.fc2.weight,
-    )
-
-    # Set up optimizers
-    lr = 1e-3
-    beta1, beta2, beta3 = 0.9, 0.999, 0.9999
-    eps = (1e-30, 1e-16)
-    torch_optim = CAME(setup_param_groups(torch_model), lr=lr, betas=(beta1, beta2, beta3), eps=eps)
-    optim = DistributedCAME(
-        setup_param_groups(tp_model),
-        lr=lr,
-        betas=(beta1, beta2, beta3),
-        eps=eps,
-    )
+    torch.set_default_dtype(dtype)
+    # set_seed(42)
+
+    # ==============================
+    # Model Init
+    # ==============================
+    base_model = MlpModel().to(local_rank)
+    tp_model = copy.deepcopy(base_model).to(local_rank)
+
+    base_param_group = setup_param_groups(base_model)
+    tp_param_group = setup_param_groups(tp_model)
+
+    # ==============================
+    # Optimizer Init
+    # ==============================
+    base_optim = CAME(base_param_group, lr=1e-3)
+    dist_optim = DistributedCAME(tp_param_group, lr=1e-3)
 
     # Setup distributed optimizer
     if zero_size > 1:
-        optim = LowLevelZeroOptimizer(
-            optim,
+        dist_optim = LowLevelZeroOptimizer(
+            dist_optim,
             overlap_communication=True,
             initial_scale=128,
             partition_grad=True,
             dp_process_group=dp_group,
             verbose=True,
         )
-        shard_to_param = optim._param_store.master_to_working_param
-        optim.optim.setup_distributed(tp_group, dp_group, shard_to_param, zero_flag=True)
+        shard_to_param = dist_optim._param_store.master_to_working_param  # {id(): param tensor} but flattened
+        dist_optim.optim.setup_distributed(
+            tensor_parallel_group=tp_group,
+            data_parallel_group=dp_group,
+            shard_to_param=shard_to_param,
+            use_zero=use_zero,
+        )
     else:
-        optim.setup_distributed(tp_group, dp_group, None, zero_flag=False)
+        shard_to_param = set_master_param_to_shard_param(tp_param_group)
+        dist_optim.setup_distributed(
+            tensor_parallel_group=tp_group,
+            data_parallel_group=dp_group,
+            shard_to_param=shard_to_param,
+            use_zero=use_zero,
+        )
```

Comment:
Low level api should not be called when using booster

Reply:
> Low level api should not be called when using booster

Fix in https://github.com/hpcaitech/ColossalAI/pull/5676/commits/74d4facc625e236d232b3ae845431537c9e08748；